sdk/containerregistry/container-registry/src/generated/generatedClientContext.ts

```diff
@@ -27,17 +27,24 @@ export class GeneratedClientContext extends coreClient.ServiceClient {
     if (!options) {
       options = {};
     }
-
     const defaults: GeneratedClientOptionalParams = {
       requestContentType: "application/json; charset=utf-8"
     };
 
+    const packageDetails = `azsdk-js-container-registry/1.0.0-beta.4`;
+    const userAgentPrefix =
+      options.userAgentOptions && options.userAgentOptions.userAgentPrefix
+        ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
+        : `${packageDetails}`;
+    options.userAgentOptions = {
+      userAgentPrefix: userAgentPrefix
```

Comment:
Were we setting the user agent string in the convenience layer before? If so, are we now ending up with duplicates?

Reply:
You are right. This would leads to duplicate of `azsdk-js-container-registry/1.0.0-beta.4`.  In convenience layer we already append library info to `options.userAgentOptions.userAgentPrefix`.

@sarangan12 what are your thoughts? Update code gen to not append packageDetails when userAgentPrefix is passed via options, or update convenice layer to not append library info?

Reply:
We were thinking of convenience layer not appending library info. From the very beginning this was one of the things that we felt could be generated. Now that we are generating, we should make use of that  :)

The only thing that should be added as a prefix to this is what the user provides

Reply:
Makes sense.  I will update the convenience layer.

Reply:
Found a bug in the codegen: my code is creating two generated clients using the same options. However, as the generated code mutates my option object, I end up having duplicate library info.  I fixed it by making an updated copy of the options bag.

```diff
-    options.userAgentOptions = {
-      userAgentPrefix: userAgentPrefix
-    };
-
     const optionsWithDefaults = {
       ...defaults,
       ...options,
+      userAgentOptions: {
+        userAgentPrefix
+      },
       baseUri: options.endpoint || "{url}"
     };
```

Reply:
Logged https://github.com/Azure/autorest.typescript/issues/1039